File: scenarios/codegen_exec/fixtures/collect_report.py
from __future__ import annotations

import hashlib
import json
import os
import subprocess
import sys
from pathlib import Path


def sha256(path: Path) -> str:
    digest = hashlib.sha256()
    digest.update(path.read_bytes())
    return digest.hexdigest()


def main() -> None:
    project_dir = Path(sys.argv[1])
    result = subprocess.run(["pytest", "-q", str(project_dir)], capture_output=True, text=True, check=True)
    payload = {
      "patched_file": os.environ.get("PATCHED_FILE", ""),
      "tests": {
        "status": "passed",
        "passed": 2,
        "stdout": result.stdout.strip()
      },
      "hashes": {
        "math_ops.py": sha256(project_dir / "math_ops.py")
      }
    }
    print(json.dumps(payload))


if __name__ == "__main__":
    main()

File: scenarios/codegen_exec/fixtures/collect_report.ts
import { createHash } from "node:crypto";
import { readFile } from "node:fs/promises";
import { spawnSync } from "node:child_process";
import path from "node:path";

function sha256(content: Buffer): string {
  return createHash("sha256").update(content).digest("hex");
}

const projectDir = process.argv[2];
if (!projectDir) {
  throw new Error("project directory is required");
}

const result = spawnSync("tsx", ["--test", path.join(projectDir, "math_ops.test.ts")], {
  encoding: "utf8",
});

if (result.status !== 0) {
  process.stderr.write(result.stderr || result.stdout);
  process.exit(result.status ?? 1);
}

const mathOpsPath = path.join(projectDir, "math_ops.ts");
const payload = {
  patched_file: process.env.PATCHED_FILE ?? "",
  tests: {
    status: "passed",
    passed: 2,
    stdout: result.stdout.trim(),
  },
  hashes: {
    "math_ops.ts": sha256(await readFile(mathOpsPath)),
  },
};

console.log(JSON.stringify(payload));
